robot/web/stream/server.go

```diff
@@ -252,14 +212,15 @@ func (srs *streamRPCServer) AddStream(ctx context.Context, req *streampb.AddStre
 	return &streampb.AddStreamResponse{}, nil
 }
 
-func (srs *streamRPCServer) RemoveStream(ctx context.Context, req *streampb.RemoveStreamRequest) (*streampb.RemoveStreamResponse, error) {
+// RemoveStream implements part of the StreamServiceServer.
+func (ss *Server) RemoveStream(ctx context.Context, req *streampb.RemoveStreamRequest) (*streampb.RemoveStreamResponse, error) {
 	pc, ok := rpc.ContextPeerConnection(ctx)
 	if !ok {
 		return nil, errors.New("can only remove a stream over a WebRTC based connection")
 	}
 
 	var streamToRemove *streamState
-	for _, stream := range srs.ss.streams {
+	for _, stream := range ss.streams {
```

Comment:
Same question regarding whether or not we need to lock `mu` before accessing `ss.streams`

Reply:
I've moved the "write lock" up to include this.